File: LeetCode/Bull and Cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        unordered_map<char, int> m;
        
        int bulls = 0, cows = 0;
        
        for(int i=0; i < guess.size(); i++) {
            if (secret[i] == guess[i])
                bulls++;
            else 
                m[guess[i]]++;
        }
        
        for(int i=0; i < secret.size(); i++) {
            if (secret[i] == guess[i])
                continue;
            else if (m[secret[i]]) {
                cows++;
                m[secret[i]]--;
            }
        }
        
        return to_string(bulls) + "A" + to_string(cows) + "B";
    }
};

File: LeetCode/Bull and Cows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        bulls, cows = 0, 0
        hashmap = {}
        
        for i in range(len(guess)):
            if secret[i] == guess[i]:
                bulls += 1
            else:
                hashmap[secret[i]] = hashmap.get(secret[i], 0) + 1
        
        for i in range(len(guess)):
            if secret[i] == guess[i]:
                continue
            elif hashmap.get(guess[i], 0) != 0:
                cows += 1
                hashmap[guess[i]] -= 1
            
        return f'{bulls}A{cows}B'